ui-spacetimechart/src/hooks/useCanvas.ts

```diff
@@ -61,7 +61,9 @@ export function useCanvas(
         const set = pickingFunctions.current[layer];
 
         if (ctx) {
-          ctx.clearRect(0, 0, ctx.canvas.width, ctx.canvas.height);
+          const scale = window.devicePixelRatio || 1;
+          ctx.clearRect(0, 0, ctx.canvas.width / scale, ctx.canvas.height / scale);
```

Comment:
Could we use `size` instead here? This would avoid duplicating the `devicePixelRatio` logic in case we update how the size is computed in the future.

Alternatively, we could reset the transform:

```ts
ctx.save();
ctx.setTransform(1, 0, 0, 1, 0, 0); // identity
ctx.clearRect(0, 0, canvas.width, canvas.height);
ctx.restore();
```